### DISCUSSION TITLE
Is it possible to have an automation which creates a new table?

### DISCUSSION THREAD

**Discussion by swing18** [Budibase/budibase#5936] [Help] [answered]
Was hoping to have a Projects table which when a row is added with project information, it would create a new project page/table. When created, the user could then add their own data to the project table. Wanted to know if this was possible and if so how would it be done. Thanks!

**Comment by melohagan** [comment, marked as answer]
Hi @swing18 

This should be possible. Your automation would look as so:

1. Row Created (Trigger for Project Information)
2. External Data Connector (Create new table using the API: https://docs.budibase.com/reference/post_tables)

Currently I don't believe there's a way to set table specific permissions, but this may be worked on in future.

**Reply by swing18** [reply]
Thanks @melohagan! I'm new to the platform so I was wondering how you would make an API request from within the automation section as the External Data Connector only has a drop down list of Queries and an add looping button.

**Reply by melohagan** [reply]
You can create a REST query, which will be available in the list of queries: https://docs.budibase.com/docs/rest-queries

**Reply by Nasnl** [reply]
@swing18 And to add to that (because I was searching a lot before I found that too): if you wonder where you can add that query:

- Add a datasource, choose for REST and then you can add your query

**Reply by swing18** [reply]
I'm still having some trouble and some more help would be greatly appreciated! Using this link: https://docs.budibase.com/reference/post_tables, I've successfully created tables in my source’s column on budibase with my desired column titles after inputting my API key and app ID and clicking "try it".

However, I haven't been able to replicate this within budibase with a REST query. I input https://budibase.app/api/public/v1/tables as my endpoint after adding and naming a query. However, sending this query results in the following error: {"message":"Invalid body - \"schema\" is required","status":400}

I've tried playing around with the Body heading, using my columns in the form data other JSON object created from "try it" I mentioned earlier in the raw(JSON) tab but with no luck. I feel like the curl code would be useful here but I'm not sure where to use it. 

Any idea what I'm doing wrong?

**Reply by melohagan** [reply]
Hey @swing18 Could you provide a screenshot of your query body? It should look something like:
<img width="608" alt="Screenshot 2022-05-20 at 08 57 33" src="https://user-images.githubusercontent.com/101575380/169481708-91886b4d-50d9-4964-8584-a5fa250fd939.png">

**Reply by swing18** [reply]
![Screenshot 2022-05-20 103134](https://user-images.githubusercontent.com/60357645/169550258-2782fe95-9fe7-49a1-bc54-9ab66c349d07.png)

**Reply by melohagan** [reply]
Great, thanks. The URL needs to be your domain. So if it's in the cloud it'll be `https://mydomain.budibase.app/api/public/v1/tables`

And of course populate the table id.

If you're running locally the URL would be: `localhost:10000/api/public/v1/tables`

**Reply by swing18** [reply]
Using https://swing3.budibase.app/api/public/v1/tables prompts the same error. Is there soemthing else I've done wrong? The table ID i'm using is the one generated with the JSON code here https://docs.budibase.com/reference/post_tables

**Reply by melohagan** [reply]
So you're still getting the error "schema required"? Maybe try removing the data section so that the payload appears as:
```
{
  "_id": "...",
  ....
  "schema": { ... }
}
```

**Reply by swing18** [reply]
updated to:
{
  
    "_id": "ta_350.....",
    "name": "New Table",
    "schema": {
   
    }
}
Got a new error message says this:
{
  "message": "Invalid app ID provided, please check the x-budibase-app-id header.",
  "status": 400
}

**Reply by melohagan** [reply]
OKay, you'll need to provide the app-id header.
![Screenshot 2022-05-20 at 16 52 43](https://user-images.githubusercontent.com/101575380/169565796-53fbb31b-ed1f-42b2-9eb1-d39811afebaa.png)

Click on the headers tab and fire it in. You should be able to see the app id in the URL of your Budibase builder. e.g. app_...

**Reply by swing18** [reply]
![image](https://user-images.githubusercontent.com/60357645/169568451-712d8c92-f947-492a-822d-05e6374e3f55.png)

same error with this. So strangethat using all of my parameters on this link: https://docs.budibase.com/reference/post_tables and clicking try it creates a table in my budibase page, but this query within my budibase page won't.

**Reply by melohagan** [reply]
the key is `x-budibase-app-id`, the value is `app_dev...`

**Reply by swing18** [reply]
Tried that, now getting:
Query Error: maximum redirect reached at: https://XXX.budibase.app/builder

**Reply by swing18** [reply]
Wanted to come back to this and mention I got the query to work by using additional headers. Thanks for the help!
